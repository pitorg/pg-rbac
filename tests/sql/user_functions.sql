-- The user_id is generated outside of the rbac extension
SELECT 'd48572ae-6087-432a-927b-0c6604b9d84d'::UUID AS user_id \gset
SELECT rbac.add_user(:'user_id'::UUID);
SELECT rbac.add_user(:'user_id'::UUID); -- error since already added
SELECT rbac.add_user(NULL::UUID); -- error since NULL is invalid
SELECT rbac.delete_user(NULL::UUID); -- NULL since NULL is invalid
SELECT rbac.delete_user(:'user_id'::UUID);
SELECT rbac.delete_user(:'user_id'::UUID); -- NULL since already deleted
SELECT rbac.delete_user('fd939c0d-728e-46ff-a535-58464c20917a'::UUID); -- NULL since non-existing
