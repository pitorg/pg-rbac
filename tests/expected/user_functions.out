-- The user_id is generated outside of the rbac extension
SELECT 'd48572ae-6087-432a-927b-0c6604b9d84d'::UUID AS user_id \gset
SELECT rbac.add_user(:'user_id'::UUID);
 add_user 
----------
 t
(1 row)

SELECT rbac.add_user(:'user_id'::UUID); -- error since already added
ERROR:  duplicate key value violates unique constraint "users_pkey"
DETAIL:  Key (id)=(d48572ae-6087-432a-927b-0c6604b9d84d) already exists.
CONTEXT:  SQL function "add_user" statement 1
SELECT rbac.add_user(NULL::UUID); -- error since NULL is invalid
ERROR:  null value in column "id" of relation "users" violates not-null constraint
DETAIL:  Failing row contains (null).
CONTEXT:  SQL function "add_user" statement 1
SELECT rbac.delete_user(NULL::UUID); -- NULL since NULL is invalid
 delete_user 
-------------
 
(1 row)

SELECT rbac.delete_user(:'user_id'::UUID);
 delete_user 
-------------
 t
(1 row)

SELECT rbac.delete_user(:'user_id'::UUID); -- NULL since already deleted
 delete_user 
-------------
 
(1 row)

SELECT rbac.delete_user('fd939c0d-728e-46ff-a535-58464c20917a'::UUID); -- NULL since non-existing
 delete_user 
-------------
 
(1 row)

